src/decoders/cr2.rs

```diff
@@ -142,7 +142,8 @@ impl<'a> Decoder for Cr2Decoder<'a> {
       }
     };
 
-    let mut img = RawImage::new(camera, width, height, try!(self.get_wb(camera)), image);
+    let wb = self.get_wb(&camera)?;
+    let mut img = RawImage::new(camera, width, height, wb, image);
```

Comment:
This seems like an unrelated code style change, no? Or do you need wb later for some reason?